example/app/composeApp/src/commonMain/kotlin/composegears/tiamat/example/ViewModels.kt

```diff
@@ -76,25 +84,23 @@ val ViewModelsScreen1 by navDestination<Unit> {
 
 val ViewModelsScreen2 by navDestination<Unit> {
     val navController = navController()
-    val sharedViewModel = rememberSharedViewModel { SharedViewModel() }
+    val sharedViewModel = rememberSharedViewModel(provider = ::SharedViewModel)
 
     SimpleScreen("ViewModel's - Screen 2") {
         Column(
             modifier = Modifier.padding(16.dp),
             horizontalAlignment = Alignment.CenterHorizontally
         ) {
             val sharedTimerValue by sharedViewModel.timer.collectAsState()
-            ViewModelInfo(hashCode = sharedViewModel.hashCode(), timer = sharedTimerValue)
+            ViewModelInfo("SharedViewModel", hashCode = sharedViewModel.hashCode(), timer = sharedTimerValue)
```

Comment:
can be

```kotlin
ViewModelInfo(
      name = "SharedViewModel", 
      hashCode = sharedViewModel.hashCode(), 
      timer = sharedTimerValue
)
```

Reply:
https://github.com/ComposeGears/Tiamat/pull/68#discussion_r1773678643